src/api/app/controllers/webui/user_controller.rb

```diff
@@ -156,6 +156,11 @@ def tokens
     render json: User.autocomplete_token(params[:q])
   end
 
+  def acknowledge_announcement
+    announcement = Announcement.find(params[:announcement_id])
+    User.current.acknowledge_announcement!(announcement) if announcement.present?
```

Comment:
IMO, `if announcement.present?` is not needed, because the line above will raise an exception if the object doesn't exist in the DB.